includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -211,6 +211,22 @@ public function custom_button_selector() {
 		return $this->is_custom_button() ? '#wcpay-custom-button' : '';
 	}
 
+	/**
+	 * Gets the product total price.
+	 *
+	 * @param object $product WC_Product_* object.
+	 * @return integer Total price.
+	 */
+	public function get_product_price( $product ) {
+		$product_price = $product->get_price();
+		// Add subscription sign-up fees to product price.
+		if ( 'subscription' === $product->get_type() && class_exists( 'WC_Subscriptions_Product' ) ) {
```

Comment:
I agree this doesn't look good. Hopefully we will improve this with https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1563

Reply:
Yes 👍